src/platform_impl/macos/view.rs

```diff
@@ -504,68 +613,46 @@ extern "C" fn insert_text(this: &Object, _sel: Sel, string: id, _replacement_ran
         let state_ptr: *mut c_void = *this.get_ivar("winitState");
         let state = &mut *(state_ptr as *mut ViewState);
 
-        let has_attr: BOOL = msg_send![string, isKindOfClass: class!(NSAttributedString)];
-        let characters = if has_attr != NO {
-            // This is a *mut NSAttributedString
-            msg_send![string, string]
+        let string = id_to_string_lossy(string);
+
+        let is_control = if let Some(ch) = string.chars().next() {
+            ch.is_control()
         } else {
-            // This is already a *mut NSString
-            string
+            false
         };
 
-        let slice =
-            slice::from_raw_parts(characters.UTF8String() as *const c_uchar, characters.len());
-        let string = str::from_utf8_unchecked(slice);
-
         // We don't need this now, but it's here if that changes.
         //let event: id = msg_send![NSApp(), currentEvent];
 
-        let mut events = VecDeque::with_capacity(characters.len());
-        for character in string.chars().filter(|c| !is_corporate_character(*c)) {
-            events.push_back(EventWrapper::StaticEvent(Event::WindowEvent {
+        // Marked text will be clear when inserting marked text.
+        if state.is_ime_enabled() && !is_control {
+            AppState::queue_event(EventWrapper::StaticEvent(Event::WindowEvent {
                 window_id: WindowId(get_window_id(state.ns_window)),
-                event: WindowEvent::ReceivedCharacter(character),
+                event: WindowEvent::IME(IME::Commit(string)),
             }));
+            state.ime_state = ImeState::Enabled;
         }
-
-        AppState::queue_events(events);
     }
     trace!("Completed `insertText`");
 }
 
 extern "C" fn do_command_by_selector(this: &Object, _sel: Sel, command: Sel) {
-    trace!("Triggered `doCommandBySelector`");
-    // Basically, we're sent this message whenever a keyboard event that doesn't generate a "human readable" character
-    // happens, i.e. newlines, tabs, and Ctrl+C.
+    trace!("Triggered `doCommandBySelector`, with {}", command.name());
+    // Basically, we're sent this message whenever a keyboard event was sent to
```

Comment:
```suggestion
    // Basically, we sent this message whenever a keyboard event was sent to
```

Reply:
The original was correct.

"we are sent this message..." means "this message is sent to us..."

Maybe the wording should be changed so that it's more clear.

Reply:
I changed the wording